torchgeo/datasets/spacenet.py

```diff
@@ -18,84 +20,90 @@
-    * No. of polygons - 382,534 building labels
-    * Area Coverage - 2544 sq km
-
-    Dataset format:
-
-    * Imagery - Raw 8 band Worldview-3 (GeoTIFF) & Pansharpened RGB image (GeoTIFF)
-    * Labels - GeoJSON
-
-    If you are using data from SpaceNet in a paper, please cite the following paper:
+class SpaceNet(VisionDataset, abc.ABC):
+    """Abstract base class for the SpaceNet datasets.
 
-    * https://arxiv.org/abs/1807.01232
-
-    .. note::
+    The `SpaceNet <https://spacenet.ai/datasets/>`_ datasets are a set of
+    datasets that all together contain >11M building footprints and ~20,000 km
+    of road labels mapped over high-resolution satellite imagery obtained from
+    Worldview-2 and Worldview-3 sensors.
+    """
 
-       This dataset requires the following additional library to be installed:
+    @property
+    @abc.abstractmethod
+    def dataset_id(self) -> str:
+        """Dataset ID."""
 
-       * `radiant-mlhub <https://pypi.org/project/radiant-mlhub/>`_ to download the
-         imagery and labels from the Radiant Earth MLHub
+    @property
+    @abc.abstractmethod
+    def imagery(self) -> Dict[str, str]:
+        """Mapping of image identifier and filename."""
 
-    """
+    @property
+    @abc.abstractmethod
+    def label_glob(self) -> str:
+        """Label filename."""
 
-    dataset_id = "spacenet1"
-    md5 = "e6ea35331636fa0c036c04b3d1cbf226"
-    imagery = {"rgb": "RGB.tif", "8band": "8Band.tif"}
-    label_glob = "labels.geojson"
-    foldername = "sn1_AOI_1_RIO"
+    @property
+    @abc.abstractmethod
+    def collection_md5_dict(self) -> Dict[str, str]:
+        """Mapping of collection id and md5 checksum."""
 
     def __init__(
         self,
         root: str,
-        image: str = "rgb",
+        image: str,
+        collections: Optional[List[str]] = None,
         transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
         download: bool = False,
         api_key: Optional[str] = None,
         checksum: bool = False,
     ) -> None:
-        """Initialise a new SpaceNet 1 Dataset instance.
+        """Initialise a new SpaceNet Dataset instance.
 
         Args:
             root: root directory where dataset can be found
-            image: image selection which must be "rgb" or "8band"
+            image: image selection
+            collections: collection selection
             transforms: a function/transform that takes input sample and its target as
                 entry and returns a transformed version.
             download: if True, download dataset and store it in the root directory.
             api_key: a RadiantEarth MLHub API key to use for downloading the dataset
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
 
         Raises:
             RuntimeError: if ``download=False`` but dataset is missing
         """
         self.root = root
         self.image = image  # For testing
+
+        if collections is None:
```

Comment:
Is there any reason to make it optional? We could instead default to the empty list.

Reply:
Defaulted to empty list